src/lava/magma/core/model/py/ports.py

```diff
@@ -14,31 +14,80 @@
 
 
 class AbstractPyPort(AbstractPortImplementation):
+    """Abstract class for ports implemented in python. A port must have one or
+    multiple csp ports."""
     @property
     @abstractmethod
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """
+        Abstract property to get all csp ports of a port.
+
+        Returns
+        -------
+        A list of all csp ports used by the port.
+        """
         pass
 
 
-class AbstractIOPyPort(AbstractPyPort):
+class AbstractPyIOPort(AbstractPyPort):
+    """Abstract class of an input/output port implemented in python.
+
+    Parameters
+    ----------
+    csp_ports : list
+        A list of csp ports used by this IO port.
+
+    process_model : AbstractProcessModel
+        The process model used by the process of the port.
+
+    shape : tuple, default=tuple()
+        The shape of the port.
+
+    d_type: type, default=int
+        The data type of the port.
+
+    Attributes
+    ----------
+    _csp_ports : list
+        A list of csp ports used by this IO port.
+    """
     def __init__(self,
                  csp_ports: ty.List[AbstractCspPort],
                  process_model: AbstractProcessModel,
                  shape: ty.Tuple[int, ...] = tuple(),
                  d_type: type = int):
+
         self._csp_ports = csp_ports
         super().__init__(process_model, shape, d_type)
 
     @property
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """Property to get all csp ports used by the port.
+
+        Returns
+        -------
+        A list of all csp ports used by the port.
+        """
         return self._csp_ports
 
 
-class PyInPort(AbstractIOPyPort):
+class PyInPort(AbstractPyIOPort):
     """Python implementation of InPort used within AbstractPyProcessModel.
     If buffer is empty, recv() will be blocking.
```

Comment:
What buffer?

Reply:
Is this actually true? I thought we made it non-blocking in the absence of a DONE token.

By the way, we need a DONE token to be implemented in our CSP channel infrastructure.

Reply:
recv is still blocking afaik, if there is no data. Each send needs a recv and vice versa in order to not block execution.

I removed this comment in the class description (it was a leftover from before) and added the information in the docstring of recv().

Reply:
Sorry. Now I remember, recv() is blocking when connected but when there is not data. However, if the port is not connected then it should not be blocking.